standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -242,22 +244,19 @@ public TxnHandler() {
 
   /**
    * This is logically part of c'tor and must be called prior to any other method.
-   * Not physically part of c'tor due to use of relfection
+   * Not physically part of c'tor due to use of reflection
    */
-  public void setConf(HiveConf conf) {
+  public void setConf(Configuration conf) {
     this.conf = conf;
 
     checkQFileTestHack();
 
     synchronized (TxnHandler.class) {
       if (connPool == null) {
-        //only do this once per JVM; useful for support
```

Comment:
This must be present.  It's critical for debugging in the field

Reply:
If we're going to do this it seems that it should be done in the config file itself.  It doesn't make sense to bury here in the transaction stuff.  And it seems we should make it configurable in case people don't want to pollute their log files with it.

Reply:
I've ran into this multiple times where someone claims that they have certain config when in reality there are multiple hive-conf.xml floating in the system with different settings. I agree this is not perfect but it should have a replacement before it's removed.  Otherwise it's a supportability issue.

Reply:
So are you ok with moving it to MetastoreConf.java when the config file is first instantiated?  Or are you concerned that different parts of the code might actually have different conf objects?

Reply:
dumpConfig() will print where it's reading the file from.  I think once per VM is sufficient - it doesn't matter which class it's in.